src/Pact/Eval.hs

```diff
@@ -66,15 +71,16 @@ import Pact.Gas
 
 
 evalBeginTx :: Info -> Eval e ()
-evalBeginTx i = beginTx i =<< view eeTxId
+evalBeginTx i = revokeAllCapabilities >> view eeTxId >>= beginTx i
 {-# INLINE evalBeginTx #-}
 
 evalRollbackTx :: Info -> Eval e ()
-evalRollbackTx = void . rollbackTx
+evalRollbackTx i = revokeAllCapabilities >> void (rollbackTx i)
 {-# INLINE evalRollbackTx #-}
 
 evalCommitTx :: Info -> Eval e [TxLog Value]
 evalCommitTx i = do
+  revokeAllCapabilities
```

Comment:
i'm wondering why we have to revoke capabilities rather than all transactions starting with no capabilities.

Reply:
I read it as a conservative move, which is probably best in the case where we're messing with permissions. It's cheap to reset capabilities.